app/src/main/java/org/groundplatform/android/ui/map/gms/ClippingTileProvider.kt

```diff
@@ -22,33 +22,25 @@ import com.google.android.gms.maps.model.TileProvider
 import com.google.android.gms.maps.model.TileProvider.NO_TILE
 import org.groundplatform.android.ui.map.gms.mog.ImageEditor
 import org.groundplatform.android.ui.map.gms.mog.TileCoordinates
-import org.groundplatform.android.ui.map.gms.mog.toPixelBounds
-import org.groundplatform.android.ui.map.gms.mog.toPixelCoordinate
-
-private const val MAX_ZOOM = 19
+import org.groundplatform.android.ui.map.gms.mog.pixelToLatLng
 
 class ClippingTileProvider(
   private val sourceTileProvider: TileProvider,
-  clipBounds: List<LatLngBounds>,
+  private val clipBounds: List<LatLngBounds>,
 ) : TileProvider {
 
-  private val pixelBounds = clipBounds.map { it.toPixelBounds(MAX_ZOOM) }
-
   override fun getTile(x: Int, y: Int, zoom: Int): Tile {
     val sourceTile = sourceTileProvider.getTile(x, y, zoom) ?: NO_TILE
-    if (sourceTile == NO_TILE) return sourceTile
-    // We assume if a tile is returned by the source provider that at least some pixels are within
-    // the clip bounds, so there's no need to optimize by checking before clipping.
-    return clipToBounds(TileCoordinates(x, y, zoom), sourceTile)
-  }
+    val data = sourceTile.data ?: return NO_TILE
 
-  private fun clipToBounds(tileCoords: TileCoordinates, tile: Tile): Tile {
-    if (tile.data == null) return NO_TILE
+    val coords = TileCoordinates(x, y, zoom)
     val output =
-      ImageEditor.setTransparentIf(tile.data!!) { _, x, y ->
-        val pixelCoords = tileCoords.toPixelCoordinate(x, y)
-        pixelBounds.none { it.contains(pixelCoords) }
+      ImageEditor.setTransparentIf(data) { _, x, y ->
+        // Keep only the pixels inside any clip bound — transparent elsewhere.
+        val latLng = coords.pixelToLatLng(x, y)
+        val insideAny = clipBounds.any { it.contains(latLng) }
+        !insideAny // transparent if outside all clip bounds
```

Comment:
Still open from previous review. Can you please explain why we needed `pixelToLatLng` and existing `toPixelCoordinate` wasn't enough?